src/client.ts

```diff
@@ -33,12 +36,25 @@ export const add = async (
   name?: string
 ): Promise<CID> => {
   const headers = { 'content-type': 'application/octet-stream' }
+  const maxContentLength = MAX_CONTENT_LENGTH;
   const nameStr = name ? `?name=${name}` : ''
-  const { data } = await axios.post<CID>(`${baseURL}/ipfs${nameStr}`, content, {
+  const axiosOptions = {
     headers,
+    maxContentLength,
     auth
-  })
-  return data
+  }
+  
+  try {
+    const { data } = await axios.post<CID>(`${baseURL}/ipfs${nameStr}`, content, axiosOptions)
+    return data
+  } catch (err) {
+    switch (err.message) {
```

Comment:
We shouldn't switch on the error message itself. It's both brittle and subtle. i.e. Axios could change this message and it wouldn't warrant a major or even minor version increase. This means that we could apply the latest patch containing the change and this branch of our logic would never be hit again.

Reply:
Okay, so I looked into this one and it turns out to be a little challenging.  Here's the problem:
```js
Object.getOwnPropertyNames(axiosError)
// [ 'stack', 'message', 'config', 'request', 'response', 'isAxiosError', 'toJSON' ]
```
Looking through the props, I can't see any other way to distinguish between errors other than message.  

What we could do is add the following to the test suite:
```js
describe("Axios Errors", () => {
  it("returns specific error message when maxContentLength exceeded", async () => {
    const axiosOptions = {
      headers: {
        'content-type': 'application/octet-stream'
      },
      maxContentLength: 1,
      auth: TEST_AUTH
    }
    expect.assertions(1)
    try {
      const { data } = await axios.post(`${BASE_URL_DEFAULT}/ipfs`, JSON_CONTENT, axiosOptions)
    } catch (axiosError) {
      expect(axiosError.message).toBe("Request body larger than maxBodyLength limit")
    }
  })
})
```
There could be some delay between releases where the problem goes unoticed, but if the fall back is to `throw axiosError`, it's exactly the same as not handling it.

There's also a problem using `instanceof` because it can only be checked against constructors (`class` or `interface`).  Unless there's a way around checking against the message, we end up with this: 
```js
export interface AxiosFileSizeError extends Error {
    config: {
        headers: {
            "Content-Length": ByteLength
        }
        maxContentLength: ByteLength
    }
}

export class AxiosFileSizeError extends Error {
    message: string = "Request body larger than maxBodyLength limit"
}
```